### PR TITLE
Paper corrections

### PR DESCRIPTION
**Pull request by Twon**

### CODE REVIEW

File: documentation/p1950.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <pre class='metadata'>
-Title:  <code>indirect_value</code>: An indirect value-type for C++
+Title:  <code>indirect_value</code>: A Free-Store Allocated Value Type For C++
```
Comment (jbcoe) [inline documentation/p1950.md]: Should this be free-store-allocated ?

File: documentation/p1950.md [hunk 2]
```diff
@@ -16,20 +16,13 @@ Editor: Antony Peacock, ant.peacock@gmail.com
...
-
-## Motivation
-
-The C++ standard includes support for numerous smart pointer templates, with support for new addition ongoing [Meneide et al].  One common use for these types is as the pointer-to-implementation in the Pimpl idiom [H.Sutter 11]. If the application design requires deep-copy semantic when the type is moved, existing pointer types, <code>std::unique_ptr</code> and <code>std::shared_ptr</code> are not the correct tools for the job.
-
-One option for value semantics is the use of <code>std::experimental::polymophic_value</code>. However, <code>polymophic_value</code> provides a solution to the *polymorphic copying problem*, which requires the use of type-erasure.  Such implementation details incur a cost. However, use of the Pimpl idiom will likely not involve polymorphic types.  As such using <code>polymophic_value</code> means paying for features that are not used.
+Add a class template, indirect_value, to the C++ Standard Library to support free-store allocated objects with value-like semantics.
```
Comment (jbcoe) [inline documentation/p1950.md]: Should this be free-store-allocated?

File: documentation/p1950.md [hunk 3]
```diff
@@ -39,58 +32,42 @@ public:
...
-    return *this;
-}
+#### copies
+The copy-constructor and copy-assignment operator of `std::unique_ptr` are deleted. Users of a class with a `std::unique_ptr` member will be required to implement copy-construction and copy-assignment.  Implementation of these methods for a class with PImpl members is non-trivial [S. Meyers].
 
-```
-However, this is boilerplate code that must be repeated per Pimpl implementation.  Worse still we are not able to benefit from the rule-of-zero to have the compiler generate the methods for use.
+### Issues with `polymorphic_value<impl>` implementation of PImpl Idiom
+The class template `polymorphic_value` confers value-like semantics on a free-store allocated object. This satisfies our
+requirement for const-propagation and compiler-generated copy constructors but incurs an additional cost in supporting polymorphism.
```
Comment (jbcoe) [inline documentation/p1950.md]: Prefer “a compiler generated copy constructor and copy assignment operator”

File: documentation/p1950.md [hunk 4]
```diff
@@ -39,58 +32,42 @@ public:
...
+To implement deep copies of subclasses, `polymorphic_value` employs type-erasure which has run-time cost through
```
Comment (jbcoe) [inline documentation/p1950.md]: replace “through” with “due to”?

File: documentation/p1950.md [hunk 5]
```diff
@@ -39,58 +32,42 @@ public:
...
+virtual dispatch and memory indirection. We do not need to support polymorphism for our PImpl object and should not pay for what we do not use.
 
-Using the proposed indirect_value Pimpl classes could be implemented to take advantage of the rule-of-zero to provide compiler-generated method implementations.  Note this is implementations only, function declarations are required in the header and in the source file with the <code>default</code> keyword in-order to instruct the compiler to generate the function bodies in the source translation unit where the definition of the implementation class is visible to the compiler, as discussed in [S. Meyers].   
+### Perfect Implementation of the PImpl Idiom
+Like `polymorphic_value`, `indirect_value` confers value-like semantics on a free-store allocated object. Unlike `polymorphic_value`,
```
Comment (jbcoe) [inline documentation/p1950.md]: prefer “free-store-allocated”

Comment (Twon) [reply documentation/p1950.md]: Done

File: documentation/p1950.md (hunk 2 above)
Comment (Twon) [reply documentation/p1950.md]: Done

File: documentation/p1950.md (hunk 3 above)
Comment (Twon) [reply documentation/p1950.md]: Done

File: documentation/p1950.md (hunk 4 above)
Comment (Twon) [reply documentation/p1950.md]: Done

**Pull request merged**